megamek/src/megamek/client/ui/swing/CustomMechDialog.java

```diff
@@ -285,14 +285,14 @@ public CustomMechDialog(ClientGUI clientgui, Client client, List<Entity> entitie
         boolean eligibleForOffBoard = true;
         
         for (Entity e : entities) {
-            isAero &= e instanceof Aero && !(e instanceof SmallCraft || e instanceof Jumpship);
-            isMech &= e instanceof Mech;
-            isShip &= e instanceof SmallCraft || e instanceof Jumpship;
-            isVTOL &= e instanceof VTOL;
-            isWiGE &= e instanceof Tank && e.getMovementMode() == EntityMovementMode.WIGE;
-            isQuadVee &= e instanceof QuadVee;
-            isLAM &= e instanceof LandAirMech;
-            isGlider &= e instanceof Protomech && e.getMovementMode() == EntityMovementMode.WIGE;
+            isAero &= (e instanceof Aero && !(e instanceof SmallCraft || e instanceof Jumpship));
+            isMech &= (e instanceof Mech);
+            isShip &= (e instanceof SmallCraft) || (e instanceof Jumpship);
+            isVTOL &= (e instanceof VTOL);
+            isWiGE &= (e instanceof Tank && e.getMovementMode() == EntityMovementMode.WIGE);
```

Comment:
Should be `(e instanceof Tank) && (e.getMovementMode() == EntityMovementMode.WIGE);`